### DISCUSSION TITLE
Baremetal support for NVMe?

### DISCUSSION THREAD

**Discussion by josh25219** [polarfire-soc/polarfire-soc-documentation#463] [Bare metal embedded software]
I have a client who would like to implement NVMe on PolarfireSOC using baremetal.   Do we provide drivers for this currently, or is there any plan to do so?    Do we recommend a 3rd party that has IP to support this for Baremetal?

Thanks
Josh

**Comment by hughbreslin** [comment]
There is an old PCIe bare metal driver in the bare metal examples repository but this isn't being maintained so its a bit out of date. For NVMe its assumed this would be done with an OS like Linux, we do have the older PCIe driver but no driver for the NVMe. The PolarFire PCIe should be fine to use as an interface so theres no need for any other IP, the main issue will be drivers.
